otherlibs/dune-rpc/private/dune_rpc_private.ml

```diff
@@ -266,7 +266,12 @@ module Client = struct
 
     type t =
       { chan : Chan.t
-      ; requests : (Id.t, Response.t Fiber.Ivar.t) Table.t
+      ; requests :
+          ( Id.t
+          , [ `Cancelled
+            | `Pending of [ `Ok of Response.t | `Cancelled ] Fiber.Ivar.t
+            ] )
```

Comment:
What's the difference between `` `Cancelled`` and `` `Pending `Cancelled ``?

Reply:
The difference is as follows:

* `Cancelled` is used to remember that a request was cancelled and ignore the response corresponding to the request
* `Pending ivar` is filled as `Cancelled` to signal whoever made the request that a response from the server will not arrive. The caller already knows that - since it was he who asked for the cancellation, but we don't know that.